### ISSUE TITLE
Crash on iOS 7 after updating to 7.3.0

### ISSUE THREAD

**Issue by nixxcode** [jessesquires/JSQMessagesViewController#1604]
After updating from version 7.2.0 to 7.3.0 of this project, a crash occurs on iOS 7 when a non-empty message view controller is pushed. This issue was never seen in version 7.2.0 or on iOS 9 with the latest version.

The crash is due to infinite recursion;

```
Crashed: com.apple.main-thread
0   Foundation                     0x2e8d6c90 -[NSISEngine _flushPendingRemovals] + 27
1   Foundation                     0x2e8d6b5d -[NSISEngine withBehaviors:performModifications:] + 260
2   Foundation                     0x2e8d6b5d -[NSISEngine withBehaviors:performModifications:] + 260
3   UIKit                          0x3088088b -[UIView(AdditionalLayoutSupport) updateConstraintsIfNeeded] + 170
4   UIKit                          0x307a1db3 -[UIView(Hierarchy) layoutBelowIfNeeded] + 334
5   Super App                      0x205167 -[JSQMessagesViewController jsq_setToolbarBottomLayoutGuideConstant:] (JSQMessagesViewController.m:928)
6   Super App                      0x2025ed -[JSQMessagesViewController viewWillLayoutSubviews] (JSQMessagesViewController.m:311)
7   UIKit                          0x3078cd17 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 314
8   QuartzCore                     0x3040a62b -[CALayer layoutSublayers] + 142
9   QuartzCore                     0x30405e3b CA::Layer::layout_if_needed(CA::Transaction*) + 350
10  UIKit                          0x307a1e8b -[UIView(Hierarchy) layoutBelowIfNeeded] + 550
11  Super App                      0x205167 -[JSQMessagesViewController jsq_setToolbarBottomLayoutGuideConstant:] (JSQMessagesViewController.m:928)
12  Super App                      0x2025ed -[JSQMessagesViewController viewWillLayoutSubviews] (JSQMessagesViewController.m:311)
13  UIKit                          0x3078cd17 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 314
14  QuartzCore                     0x3040a62b -[CALayer layoutSublayers] + 142
15  QuartzCore                     0x30405e3b CA::Layer::layout_if_needed(CA::Transaction*) + 350
16  UIKit                          0x307a1e8b -[UIView(Hierarchy) layoutBelowIfNeeded] + 550
17  Super App                      0x205167 -[JSQMessagesViewController jsq_setToolbarBottomLayoutGuideConstant:] (JSQMessagesViewController.m:928)
18  Super App                      0x2025ed -[JSQMessagesViewController viewWillLayoutSubviews] (JSQMessagesViewController.m:311)
19  UIKit                          0x3078cd17 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 314
20  QuartzCore                     0x3040a62b -[CALayer layoutSublayers] + 142
21  QuartzCore                     0x30405e3b CA::Layer::layout_if_needed(CA::Transaction*) + 350
22  UIKit                          0x307a1e8b -[UIView(Hierarchy) layoutBelowIfNeeded] + 550
23  Super App                      0x205167 -[JSQMessagesViewController jsq_setToolbarBottomLayoutGuideConstant:] (JSQMessagesViewController.m:928)
24  Super App                      0x2025ed -[JSQMessagesViewController viewWillLayoutSubviews] (JSQMessagesViewController.m:311)
25  UIKit                          0x3078cd17 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 314
26  QuartzCore                     0x3040a62b -[CALayer layoutSublayers] + 142
27  QuartzCore                     0x30405e3b CA::Layer::layout_if_needed(CA::Transaction*) + 350
28  UIKit                          0x307a1e8b -[UIView(Hierarchy) layoutBelowIfNeeded] + 550
29  Super App                      0x205167 -[JSQMessagesViewController jsq_setToolbarBottomLayoutGuideConstant:] (JSQMessagesViewController.m:928)
30  Super App                      0x2025ed -[JSQMessagesViewController viewWillLayoutSubviews] (JSQMessagesViewController.m:311)
```

**Comment by jessesquires** [comment]
Thanks @nixxcode !

We need to revert #1588

**Comment by jessesquires** [comment]
fixed by #1623
